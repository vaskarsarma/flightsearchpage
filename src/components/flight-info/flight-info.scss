.Flight-info {
  display: flex;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  margin: 10px;
}

@media screen and (min-width: 1100px) {
  .Flight-info img {
    margin-right: 4%;
  }
  .Flight-info .detail-label:nth-child(2),
  .Flight-info .detail-label:nth-child(3),
  .Flight-info .detail-label:nth-child(4),
  .Flight-info .detail-label:nth-child(5),
  .Flight-info .detail-label:nth-child(6) {
    flex: 1 1 9%;
    text-align: left;
  }
  .Flight-info Button {
    margin-left: 4%;
    flex: 1 1 10%;
  }
}

@media screen and (min-width: 601px) and (max-width: 1099px) {
  .Flight-info .detail-label:nth-child(2) {
    flex: 5 1 80%;
    text-align: end;
    margin-bottom: 15px;
  }
  .Flight-info .detail-label:nth-child(3),
  .Flight-info .detail-label:nth-child(4) {
    flex: 2 1 20%;
  }
  .Flight-info .detail-label:nth-child(5) {
    flex: 2 1 20%;
  }
  .Flight-info .detail-label:nth-child(6) {
    flex: 2 1 20%;
  }
  .Flight-info Button {
    flex: 1 1 70%;
  }
}

@media screen and (min-width: 401px) and (max-width: 600px) {
  .Flight-info .detail-label:nth-child(2) {
    flex: 5 1 80%;
    text-align: end;
    margin-bottom: 15px;
  }
  .Flight-info .detail-label:nth-child(3),
  .Flight-info .detail-label:nth-child(4) {
    flex: 2 1 33%;
  }
  .Flight-info .detail-label:nth-child(5) {
    flex: 2 1 11%;
    text-align: end;
  }
  .Flight-info .detail-label:nth-child(6),
  .Flight-info .detail-label:nth-child(7) {
    flex: 2 1 33%;
  }
  .Flight-info Button {
    flex: 2 1 15%;
  }
}

@media screen and (min-width: 231px) and (max-width: 400px) {
  .Flight-info .detail-label:nth-child(2) {
    flex: 5 1 80%;
    text-align: end;
    margin-bottom: 15px;
  }
  .Flight-info .detail-label:nth-child(3) {
    flex: 1 1 50%;
    text-align: left;
  }
  .Flight-info .detail-label:nth-child(4) {
    flex: 1 1 50%;
    text-align: end;
  }
  .Flight-info .detail-label:nth-child(5) {
    flex: 1 1 50%;
    text-align: left;
  }
  .Flight-info .detail-label:nth-child(6) {
    flex: 1 1 50%;
    text-align: end;
  }
  .Flight-info .detail-label:nth-child(7) {
    flex: 1 1 50%;
    text-align: left;
  }
  .Flight-info Button {
    flex: 1 1 50%;
  }
}

@media screen and (max-width: 230px) {
  .Flight-info .detail-label:nth-child(2),
  .Flight-info .detail-label:nth-child(3),
  .Flight-info
    .detail-label:nth-child(4)
    .Flight-info
    .detail-label:nth-child(5),
  .Flight-info .detail-label:nth-child(6),
  .Flight-info .detail-label:nth-child(7) {
    flex: 1 1 100%;
    text-align: left;
    margin-bottom: 10px;
  }
  .Flight-info Button {
    flex: 1 1 80%;
  }
}
