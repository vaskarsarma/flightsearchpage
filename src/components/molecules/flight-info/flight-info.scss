.Flight-info {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
	align-items: center;
	padding: 10px;
	margin: 10px;
}
@media screen and(min-width: 1100px) {
	.Flight-info {
		img {
			margin-right: 4%;
		}
		.detail-label {
			&:nth-child(2) {
				flex: 1 1 9%;
				text-align: left;
			}
			&:nth-child(3) {
				flex: 1 1 9%;
				text-align: left;
			}
			&:nth-child(4) {
				flex: 1 1 9%;
				text-align: left;
			}
			&:nth-child(5) {
				flex: 1 1 9%;
				text-align: left;
			}
			&:nth-child(6) {
				flex: 1 1 9%;
				text-align: left;
			}
		}
		Button {
			margin-left: 4%;
			flex: 1 1 10%;
		}
	}
}
@media screen and(min-width: 601px) and(max-width: 1099px) {
	.Flight-info {
		.detail-label {
			&:nth-child(2) {
				flex: 5 1 80%;
				text-align: end;
				margin-bottom: 15px;
			}
			&:nth-child(3) {
				flex: 2 1 20%;
			}
			&:nth-child(4) {
				flex: 2 1 20%;
			}
			&:nth-child(5) {
				flex: 2 1 20%;
			}
			&:nth-child(6) {
				flex: 2 1 20%;
			}
		}
		Button {
			flex: 1 1 70%;
		}
	}
}
@media screen and(min-width: 401px) and(max-width: 600px) {
	.Flight-info {
		.detail-label {
			&:nth-child(2) {
				flex: 5 1 80%;
				text-align: end;
				margin-bottom: 15px;
			}
			&:nth-child(3) {
				flex: 2 1 33%;
			}
			&:nth-child(4) {
				flex: 2 1 33%;
			}
			&:nth-child(5) {
				flex: 2 1 11%;
				text-align: end;
			}
			&:nth-child(6) {
				flex: 2 1 33%;
			}
			&:nth-child(7) {
				flex: 2 1 33%;
			}
		}
		Button {
			flex: 2 1 15%;
		}
	}
}
@media screen and(min-width: 231px) and(max-width: 400px) {
	.Flight-info {
		.detail-label {
			&:nth-child(2) {
				flex: 5 1 80%;
				text-align: end;
				margin-bottom: 15px;
			}
			&:nth-child(3) {
				flex: 1 1 50%;
				text-align: left;
			}
			&:nth-child(4) {
				flex: 1 1 50%;
				text-align: end;
			}
			&:nth-child(5) {
				flex: 1 1 50%;
				text-align: left;
			}
			&:nth-child(6) {
				flex: 1 1 50%;
				text-align: end;
			}
			&:nth-child(7) {
				flex: 1 1 50%;
				text-align: left;
			}
		}
		Button {
			flex: 1 1 50%;
		}
	}
}
@media screen and(max-width: 230px) {
	.Flight-info {
		.detail-label {
			&:nth-child(2) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
			&:nth-child(3) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
			&:nth-child(6) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
			&:nth-child(7) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
			&:nth-child(4) .Flight-info .detail-label:nth-child(5) {
				flex: 1 1 100%;
				text-align: left;
				margin-bottom: 10px;
			}
		}
		Button {
			flex: 1 1 80%;
		}
	}
}
