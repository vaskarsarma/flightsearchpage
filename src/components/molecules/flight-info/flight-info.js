import React from "react";
import { Card, Button } from "react-bootstrap";
import { DetailLabel } from "../../atoms/detail-label/detail-label";
import FlightLogo from "../../atoms/flightlogo/flightlogo";
import nonStopFlightLogo from "../../../assets/nonstop.png";
import "./flight-info.scss";

export const FlightInfo = (props) => {
  const { id, flightInfoObj, disableBtn, onClickedhandler } = props;
  const depratBy = flightInfoObj.date + " " + flightInfoObj.departureTime;
  const arrivedBy = flightInfoObj.date + " " + flightInfoObj.arrivalTime;
  const price = "$" + flightInfoObj.price;

  return (
    <div className="Cart-Container">
      <Card
        bg={disableBtn ? "warning" : ""}
        border={disableBtn ? "danger" : ""}
      >
        <section className="Flight-info">
          <FlightLogo
            imagePath={nonStopFlightLogo}
            alttext="Icon"
            size="Flight-Icon"
          ></FlightLogo>
          <DetailLabel
            mainText="AIRLINE"
            subText={flightInfoObj.name}
          ></DetailLabel>
          <DetailLabel
            mainText="FROM"
            subText={flightInfoObj.origin}
          ></DetailLabel>
          <DetailLabel
            mainText="TO"
            subText={flightInfoObj.destination}
          ></DetailLabel>
          <DetailLabel mainText="DEPART BY" subText={depratBy}></DetailLabel>
          <DetailLabel mainText="ARRIVED AT" subText={arrivedBy}></DetailLabel>
          <DetailLabel mainText="PRICE" subText={price}></DetailLabel>
          <Button
            id={id}
            variant="primary"
            onClick={onClickedhandler}
            disabled={disableBtn}
          >
            SELECT
          </Button>
        </section>
      </Card>
    </div>
  );
};
