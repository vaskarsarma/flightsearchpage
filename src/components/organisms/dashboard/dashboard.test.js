import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Dashboard from "./dashboard";
import { FlightInfo } from "../../molecules/flight-info/flight-info";

describe("<Dashboard />", () => {
  let container;

  beforeEach(() => (container = shallow(<Dashboard />)));

  it("<Dashboard /> Rendered Correctly", () => {
    expect(shallowToJson(container)).toMatchSnapshot();
  });

  it("should render the FlightInfo Component", () => {
    expect(container.containsMatchingElement(<FlightInfo />)).toEqual(true);
  });

  it("FlightInfo components > disableBtn props should be true if state is changed", () => {
    render(<Dashboard />);

    userEvent.click(screen.getAllByText("SELECT")[0]);

    const outputElement = screen.getAllByText("SELECT")[0];

    expect(outputElement).toHaveProperty("disabled");
  });
});
