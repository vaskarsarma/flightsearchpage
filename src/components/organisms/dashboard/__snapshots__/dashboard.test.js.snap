// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> <Dashboard /> Rendered Correctly 1`] = `
Array [
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "6:00",
        "date": "2021/11/01",
        "departureTime": "5:00",
        "destination": "Mumbai (BOM)",
        "flightNo": "EK-101",
        "name": "EMIRATES",
        "origin": "Dubai (DXB)",
        "price": 3525,
      }
    }
    id={1}
    key="0"
    onClickedhandler={[Function]}
  />,
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "9:50",
        "date": "2021/11/01",
        "departureTime": "7:20",
        "destination": "Delhi (DEL)",
        "flightNo": "EK-102",
        "name": "EMIRATES",
        "origin": "Dubai (DXB)",
        "price": 5635,
      }
    }
    id={2}
    key="1"
    onClickedhandler={[Function]}
  />,
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "7:10",
        "date": "2021/11/01",
        "departureTime": "6:10",
        "destination": "Mumbai (BOM)",
        "flightNo": "EK-103",
        "name": "EMIRATES",
        "origin": "Dubai (DXB)",
        "price": 2537,
      }
    }
    id={3}
    key="2"
    onClickedhandler={[Function]}
  />,
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "10:20",
        "date": "2021/11/01",
        "departureTime": "8:10",
        "destination": "Delhi (DEL)",
        "flightNo": "EK-104",
        "name": "EMIRATES",
        "origin": "Dubai (DXB)",
        "price": 4681,
      }
    }
    id={4}
    key="3"
    onClickedhandler={[Function]}
  />,
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "9:15",
        "date": "2021/11/01",
        "departureTime": "8:15",
        "destination": "Bengaluru (BLR)",
        "flightNo": "EK-105",
        "name": "EMIRATES",
        "origin": "Dubai (DXB)",
        "price": 3549,
      }
    }
    id={5}
    key="4"
    onClickedhandler={[Function]}
  />,
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "13:00",
        "date": "2021/11/01",
        "departureTime": "11:00",
        "destination": "Dubai (DXB)",
        "flightNo": "EK-106",
        "name": "EMIRATES",
        "origin": "Bengaluru (BLR)",
        "price": 5645,
      }
    }
    id={6}
    key="5"
    onClickedhandler={[Function]}
  />,
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "10:00",
        "date": "2021/11/01",
        "departureTime": "8:00",
        "destination": "Dubai (DXB)",
        "flightNo": "EK-107",
        "name": "EMIRATES",
        "origin": "Bengaluru (BLR)",
        "price": 5000,
      }
    }
    id={7}
    key="6"
    onClickedhandler={[Function]}
  />,
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "10:30",
        "date": "2021/11/01",
        "departureTime": "8:00",
        "destination": "Bengaluru (BLR)",
        "flightNo": "EK-108",
        "name": "EMIRATES",
        "origin": "Dubai (DXB)",
        "price": 4004,
      }
    }
    id={8}
    key="7"
    onClickedhandler={[Function]}
  />,
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "15:30",
        "date": "2021/11/01",
        "departureTime": "12:00",
        "destination": "Dubai (DXB)",
        "flightNo": "EK-109",
        "name": "EMIRATES",
        "origin": "Bengaluru (BLR)",
        "price": 3495,
      }
    }
    id={9}
    key="8"
    onClickedhandler={[Function]}
  />,
  <Memo()
    disableBtn={false}
    flightInfoObj={
      Object {
        "arrivalTime": "11:30",
        "date": "2021/11/01",
        "departureTime": "9:00",
        "destination": "Delhi (DEL)",
        "flightNo": "EK-110",
        "name": "EMIRATES",
        "origin": "Dubai (DXB)",
        "price": 4137,
      }
    }
    id={10}
    key="9"
    onClickedhandler={[Function]}
  />,
]
`;
